spec.md

```diff
@@ -1172,38 +1172,62 @@ The CO MUST implement the specified error recovery behavior when it encounters t
-This RPC call SHALL return `supported` only if all the volume capabilities specified in the request are supported.
+This RPC call SHALL return `confirmed` only if all the volume capabilities specified in the request are supported.
 This operation MUST be idempotent.
 
+NOTE: Older plugins will parse but likely not "process" newer fields that may be present in capability-validation messages (and sub-messages) sent by a CO that is communicating using a newer, backwards-compatible version of the CSI protobufs.
+Therefore, the CO SHALL reconcile successful capability-validation responses by comparing the validated capabilities with those that it had originally requested.
+
 ```protobuf
 message ValidateVolumeCapabilitiesRequest {
   // The ID of the volume to check. This field is REQUIRED.
   string volume_id = 1;
 
+  // Attributes of the volume to check. This field is OPTIONAL and MUST
+  // match the attributes of the Volume identified by `volume_id`.
+  map<string, string> volume_attributes = 2;
+
   // The capabilities that the CO wants to check for the volume. This
   // call SHALL return "supported" only if all the volume capabilities
   // specified below are supported. This field is REQUIRED.
-  repeated VolumeCapability volume_capabilities = 2;
+  repeated VolumeCapability volume_capabilities = 3;
 
-  // Attributes of the volume to check. This field is OPTIONAL and MUST
-  // match the attributes of the Volume identified by `volume_id`.
-  map<string, string> volume_attributes = 3;
+  // See CreateVolumeRequest.parameters.
+  // This field is OPTIONAL.
+  map<string, string> parameters = 4;
 
-  // Specifies where (regions, zones, racks, etc.) the caller believes
-  // the volume is accessible from.
-  // A caller MAY specify multiple topologies to indicate they believe
-  // the volume to be accessible from multiple locations.
-  // This field is OPTIONAL. This field SHALL NOT be set unless the
-  // plugin advertises the ACCESSIBILITY_CONSTRAINTS capability.
-  repeated Topology accessible_topology = 4;
+  // See CreateVolumeRequest.accessibility_requirements.
+  // This field is OPTIONAL.
+  TopologyRequirement accessibility_requirements = 5;
+
+  // Secrets required by plugin to complete volume validation request.
+  // This field is OPTIONAL. Refer to the `Secrets Requirements`
+  // section on how to use this field.
+  map<string, string> controller_validate_secrets = 6;
 }
 
 message ValidateVolumeCapabilitiesResponse {
-  // True if the Plugin supports the specified capabilities for the
-  // given volume. This field is REQUIRED.
-  bool supported = 1;
+  message Confirmed {
+    // Volume capabilities supported by the plugin.
+    // This field is REQUIRED.
+    repeated VolumeCapability volume_capabilities = 1;
+
+    // Topology requirements supported by the plugin.
+    // This field is OPTIONAL.
+    TopologyRequirement accessibility_requirements = 2;
+  }
```

Comment:
add parameters and attributes to `Confirmed`

Reply:
After looking a bit more closely, I am actually a bit confused why we need topology in this RPC at all.  Since `ListVolume` returns a list of `Volume` messages, and `Volume` contains `accessible_topology` the CO should already have all the information it requires about topology just by listing the volumes.

It makes total sense to me to include `parameters` but maybe we don't need  `accessibility_requirements`?  Or are you thinking that you don't want the CO to have to do the logic to figure out if `accessible_topology` satisfies `accessibility_requirements`

Reply:
Discussed this with @jieyu earlier today. I think we're OK with removing topology from the validation call. Would that be satisfactory?

Reply:
sounds good to me.